src/MySqlConnector/MySqlClient/MySqlConnection.cs

```diff
@@ -349,6 +354,12 @@ private void CloseDatabase()
 			}
 		}
 
+		private void SetConnectionSettings(MySqlConnectionStringBuilder connectionStringBuilder)
+		{
+			m_connectionStringBuilder = connectionStringBuilder;
```

Comment:
We need a private copy of this object, to avoid problems if the caller mutates it after passing it to the constructor.